client/src/components/Header.tsx

```diff
@@ -14,17 +15,25 @@ export function Header() {
     setMenuIsOpen(false);
   }
 
+  useEffect(() => {
+    const localAuthJson = localStorage.getItem('um.auth');
```

Comment:
Since the key `um.auth` is defined and used in data.ts, I would move this code to data.ts and put it in a separate function. That way, all your localStorage code is isolated in data.ts.